addon/components/connect.js

```diff
@@ -37,9 +38,16 @@ export default (stateToComputed, dispatchToActions=() => ({})) => {
           });
         }
 
-        this.actions = Object.assign({},
-          this.actions, dispatchToActions.call(this, redux.dispatch.bind(redux))
-        );
+        if (typeof dispatchToActions === 'function') {
+          this.actions = Object.assign({},
+            this.actions, dispatchToActions.call(this, redux.dispatch.bind(redux))
+          );
+        }
+
+        if (typeof dispatchToActions === 'object') {
+          this.actions = Object.assign({},
+            this.actions, bindActionCreators.call(this, dispatchToActions, redux.dispatch.bind(redux)));
```

Comment:
I think that makes sense. That way the context of the action creator isn't the component calling it. If you need properties from your component, they should be passed explicitly.